src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -115,6 +115,10 @@ internal void Dispose(bool joinThreads)
             {
                 // set _disposing so that we don't attempt restart the host anymore:
                 _disposing = true;
+                if (Interlocked.Exchange(ref _processExitHandling, ProcessExitHandled) == ProcessExitHooked)
```

Comment:
>if [](start = 16, length = 2)

This doesn't guarantee that ProcessExitedHandler does not call _host.OnProcesExited after we return from Dispose.

There is a race condition:
1) ProcessExitedHandler has already been called and in it _disposing already tested, yet OnProcessExited has not been called yet
2) Dispose is called
3) Dispose returns 
4) ProcessExitedHandler is resumed and calls OnProcessExited